language/stdlib/modules/genesis.move

```diff
@@ -0,0 +1,75 @@
+address 0x0:
+module Genesis {
+    use 0x0::Association;
+    use 0x0::Libra;
+    use 0x0::Event;
+    use 0x0::Coin1;
+    use 0x0::Coin2;
+    use 0x0::LBR;
+    use 0x0::AccountType;
+    use 0x0::Empty;
+    use 0x0::Unhosted;
+    use 0x0::VASP;
+    use 0x0::LibraAccount;
+    use 0x0::LCS;
+    use 0x0::TransactionFeeAccounts;
+    use 0x0::AccountTrack;
+    use 0x0::LibraConfig;
+    use 0x0::LibraWriteSetManager;
+    use 0x0::LibraBlock;
+    use 0x0::LibraTransactionTimeout;
+
+    fun initialize_accounts(
+        association_root_addr: address,
+        coin1_addr: address,
+        coin2_addr: address,
+        lbr_addr: address,
+        assoc_init_balance: u64,
+        genesis_auth_key: vector<u8>,
+    ) {
+        // Association/cap setup
+        Association::initialize();
+        Association::apply_for_privilege<Libra::AddCurrency>();
+        Association::grant_privilege<Libra::AddCurrency>(association_root_addr);
+
+        // Event and currency setup
+        Event::initialize();
+        Libra::initialize();
+        Coin1::initialize();
+        Coin2::initialize();
+        LBR::initialize();
+
+        //// Account type setup
+        AccountType::register<Unhosted::T>();
+        AccountType::register<Empty::T>();
+        VASP::initialize();
+
+        AccountTrack::initialize();
+        LibraAccount::initialize();
+        Unhosted::publish_global_limits_definition();
+        LibraAccount::create_unhosted_account<LBR::T>(
+            association_root_addr,
+            LCS::to_bytes(&association_root_addr)
+        );
+        TransactionFeeAccounts::initialize();
+        Association::grant_privilege<TransactionFeeAccounts::TransactionFeeAccountRegistration>(
+            association_root_addr
+        );
+
+        //// Register transaction fee accounts
+        LibraAccount::create_unhosted_account<Coin1::T>(coin1_addr, LCS::to_bytes(&coin1_addr));
```

Comment:
These `LCS::to_bytes` are all just hacks to create (unused, because we'll never send a tx from the tx fees account) 16 byte auth key prefixes. We might want to just do:

```
let dummy_auth_key_prefix = x"0000000000000000"; // 16 bytes of zeroes
LibraAccount::create_unhosted_account<Coin1::T>(coin1_addr, copy dummy_auth_key_prefix);
LibraAccount::create_unhosted_account<Coin2::T>(coin2_addr, copy dummy_auth_key_prefix);
LibraAccount::create_unhosted_account<LBR::T>(lbr_addr, dummy_auth_key_prefix);
```
to keep the genesis as simple as we can.